src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1884,12 +1884,31 @@ static void uct_ib_mlx5_devx_check_dp_ordering(uct_ib_mlx5_md_t *md, void *cap,
                                                void *cap_2,
                                                uct_ib_device_t *dev)
 {
+    md->dv_ooo_cap.max_dp_ordering_rc = UCT_IB_MLX5_DP_ORDERING_IBTA;
+    md->dv_ooo_cap.max_dp_ordering_dc = UCT_IB_MLX5_DP_ORDERING_IBTA;
+
     if (UCT_IB_MLX5DV_GET(cmd_hca_cap, cap, dp_ordering_ooo_rw_rc)) {
-        md->flags |= UCT_IB_MLX5_MD_FLAG_DP_ORDERING_OOO_RW_RC;
+        md->dv_ooo_cap.max_dp_ordering_rc =
```

Comment:
This implies that RW is always supported when ALL is supported.
Instantiate for RC/DC:
```
if (max_dp_ordering_dc == OOO_ALL) {
  goto done;
}

if (MLX5DV_GET(all_dc)) {
  max_dp_ordering_dc = OOO_ALL;
} else if (MLX5_DV_GET(rw_dc)) {
  max_dp_ordering_dc = RW_ALL;
} else {
  max_dp_ordering_dc = IBTA;
}
```
max is actually the most relaxed ordering so it could make sense to use cap/limit/mode wording instead.

Reply:
But it is that way, DDP cannot be enabled without AR support

Reply:
agree "This implies that RW is always supported when ALL is supported." is just side note. I still think using if/else like below could be clearer:
```
if (max_dp_ordering_dc == OOO_ALL) {
  goto done;
}

if (MLX5DV_GET(all_dc)) {
  max_dp_ordering_dc = OOO_ALL;
} else if (MLX5_DV_GET(rw_dc)) {
  max_dp_ordering_dc = RW_ALL;
} else {
  max_dp_ordering_dc = IBTA;
}
```